Modules/Core/SpatialObjects/include/itkSurfaceSpatialObject.hxx

```diff
@@ -68,18 +68,19 @@ SurfaceSpatialObject<TDimension>::InternalClone() const
 }
 
 /** Print the surface object */
-template <unsigned int TDimension>
+template <unsigned int TDimension, typename TSurfacePointType>
 void
-SurfaceSpatialObject<TDimension>::PrintSelf(std::ostream & os, Indent indent) const
+SurfaceSpatialObject<TDimension, TSurfacePointType>::PrintSelf(std::ostream & os, Indent indent) const
 {
   os << indent << "SurfaceSpatialObject(" << this << ")" << std::endl;
   Superclass::PrintSelf(os, indent);
 }
 
+#if !defined(ITK_LEGACY_REMOVE)
 /** Approximate the normals of the surface */
-template <unsigned int TDimension>
+template <unsigned int TDimension, typename TSurfacePointType>
 bool
-SurfaceSpatialObject<TDimension>::Approximate3DNormals()
+SurfaceSpatialObject<TDimension, TSurfacePointType>::Approximate3DNormals()
 {
   if (TDimension != 3)
```

Comment:
Shouldn't this method now just delegate to `ComputeNormals`?

Reply:
Done.